changefeedccl: option to load sarama config from JSON file

spiffyy99 opened the pull request:
Previously, the option `kafka_sink_config` only allowed passing
JSON over the command line. This change adds a new option
`kafka_sink_config_file` to pass a filename containing the sarama
config. Both local files and cloud storage files (S3/GCS) are
supported.

Release note: Adds option `kafka_sink_config_file` to pass sarama
config via JSON file (local/S3/GCS).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/68006)
<!-- Reviewable:end -->

amruss commented:
Are there any guardrails we need for cloud storage files we upload?

spiffyyeng commented:
> Are there any guardrails we need for cloud storage files we upload?

@amruss if you're talking about credentials and accessing the files, the libraries handle this for the most part. if `gcloud` and `aws` are installed/configured, there should be no access issues.

shermanCRL commented:
Something to think about is how we document and version the JSON schema. I guess that’s true regardless of whether it’s a file vs a SQL string.

spiffyyeng commented:
> Something to think about is how we document and version the JSON schema. I guess that’s true regardless of whether it’s a file vs a SQL string.

good point. i believe the schema just needs to be parsable into our `saramaConfig` struct (subset of the [sarama one](https://github.com/Shopify/sarama/blob/master/config.go), so we'll need to document what this looks like.

spiffyy99 committed:
changefeedccl: option to load sarama config from JSON file

Previously, the option `kafka_sink_config` only allowed passing
JSON over the command line. This change adds a new option
`kafka_sink_config_file` to pass a filename containing the sarama
config. Both local files and cloud storage files (S3/GCS) are
supported.

Release note (enterprise change): Adds option `kafka_sink_config_file`
to pass sarama config via JSON file (local/S3/GCS).
